test/new-e2e/tests/windows/install-test/persisting_integrations_test.go

```diff
@@ -0,0 +1,294 @@
+	}
+
+	// create folder in protected location with the ddagentuser as the owner
+	// run tests
+	testerOptions := []TesterOption{
+		WithAgentPackage(s.AgentPackage),
+	}
+	t, err := NewTester(s, vm, testerOptions...)
+	s.Require().NoError(err, "should create tester")
+	ddAgentUserIdentity, err := windowsCommon.GetIdentityForUser(t.host,
+		windowsCommon.MakeDownLevelLogonName(t.expectedUserDomain, t.expectedUserName),
+	)
+	s.Require().NoError(err, "should get ddagentuser identity")
+
+	// create folder owned to ddAgentUserIdenity
+	folderPath := "C:\\ProgramData\\Datadog\\protected"
+	err = vm.MkdirAll(folderPath)
+	s.Require().NoError(err, "should create folder")
+
+	// write file to folder
+	filePath := filepath.Join(folderPath, ".diff_python_installed_packages.txt")
+	_, err = vm.WriteFile(filePath, []byte(""))
+	s.Require().NoError(err, "should write file to folder")
+
+	// run powershell command to own file to ddAgentUserIdentity
+	cmd := fmt.Sprintf(`$acl = Get-Acl "%s"; $acl.SetOwner([System.Security.Principal.NTAccount]::new("%s")); Set-Acl "%s" $acl`, filePath, ddAgentUserIdentity.GetName(), filePath)
+	_, err = vm.Execute(cmd)
+	s.Require().NoError(err, "should set owner to ddAgentUserIdentity")
+
+	// upgrade to the new version, should fail due to folder permissions
+	if !s.Run(fmt.Sprintf("Install %s with failure", s.upgradeAgentPackge.AgentVersion()), func() {
+		_, err := windowsAgent.InstallAgent(vm,
+			windowsAgent.WithPackage(s.upgradeAgentPackge),
+			windowsAgent.WithInstallLogFile(filepath.Join(s.SessionOutputDir(), "upgrade.log")),
+			windowsAgent.WithIntegrationsPersistence("1"),
+		)
+		s.Require().Error(err, "should fail to install agent %s", s.upgradeAgentPackge.AgentVersion())
+	}) {
+		s.T().FailNow()
+	}
+
+	// TODO: we shouldn't have to start the agent manually after rollback
+	//       but the kitchen tests did too.
+	err = windowsCommon.StartService(vm, "DatadogAgent")
+	s.Require().NoError(err, "agent service should start after rollback")
+
+	// the previous version should be functional
+	RequireAgentVersionRunningWithNoErrors(s.T(), s.NewTestClientForHost(vm), s.AgentPackage.AgentVersion())
+
+	// Ensure services are still installed
+	// NOTE: will need to update this if we add or remove services
+	_, err = windowsCommon.GetServiceConfigMap(vm, servicetest.ExpectedInstalledServices())
+	s.Assert().NoError(err, "services should still be installed")
+
+	s.uninstallAgent()
+
+}
+
+// install third party integration
+func (s *testPersistingIntegrationsSuite) installThirdPartyIntegration(vm *components.RemoteHost, integration string) error {
+	installPath, err := windowsAgent.GetInstallPathFromRegistry(s.Env().RemoteHost)
+	s.Require().NoError(err, "should get install path from registry")
+
+	cmd := fmt.Sprintf(`& "%s\bin\agent.exe" integration install -t %s`, installPath, integration)
+	out, err := vm.Execute(cmd)
+
+	if err != nil {
+		s.T().Logf("Error installing integration %s:\n%s", integration, out)
```

Comment:
log `err` instead of `out`. on error `vm.Execute` returns empty string for `out`. But it includes `stdout` in the `err` value it returns. it's idiomatic in Go to not use the other return values when `err != nil`.